src/Lean/Server/FileWorker.lean

```diff
@@ -113,20 +118,22 @@ section Elab
     -- because empty diagnostics clear existing error/information squiggles. Therefore we always
     -- want to publish in case there was previously a message at this position.
     publishDiagnostics m snap.diagnostics.toArray ctx.hOut
+    publishIleanInfoUpdate m ctx.hOut #[snap]
     return some snap
 
   /-- Elaborates all commands after the last snap (using `headerSnap` if `snaps`
   is empty), emitting the diagnostics into `hOut`. -/
   def unfoldCmdSnaps (m : DocumentMeta) (headerSnap : Snapshot) (snaps : Array Snapshot) (cancelTk : CancelToken)
       : ReaderT WorkerContext IO (AsyncList ElabTaskError Snapshot) := do
+    let ctx ← read
     if snaps.isEmpty && headerSnap.msgLog.hasErrors then
       -- Treat header processing errors as fatal so users aren't swamped with
       -- followup errors
-      let hOut := (←read).hOut
-      publishProgressAtPos m headerSnap.beginPos hOut (kind := LeanFileProgressKind.fatalError)
+      publishProgressAtPos m headerSnap.beginPos ctx.hOut (kind := LeanFileProgressKind.fatalError)
+      publishIleanInfoFinal m ctx.hOut #[headerSnap]
       AsyncList.nil
     else
-      let ctx ← read
+      publishIleanInfoUpdate m ctx.hOut <| snaps.insertAt 0 headerSnap
```

Comment:
This will discard any previous ilean info in the watchdog because the doc version (presumably) increased, right? Could you add a small note in that regard?